server/s3_object_api_handler.cc

```diff
@@ -120,8 +121,10 @@ void S3ObjectAPIHandler::create_action() {
             action = std::make_shared<S3PutChunkUploadObjectAction>(request);
             s3_stats_inc("put_object_chunkupload_request_count");
           } else if (!request->get_header_value("x-amz-copy-source").empty()) {
-            // Copy Object not yet supported.
-            // Do nothing = unsupported API
+            request->set_action_str("CopyObject");
+            request->set_object_size(request->get_data_length());
```

Comment:
request->get_data_length() is the length of data in http body. I presume body, in case of copyobject, it is empty.
Do you want to say:
  request->set_object_size(<SourceObject.getSize()>)
Note: SourceObject.getSize() is just psedocode, and not existing in the current code.

Reply:
yes, the body (Content-Length) would be 0 for copy object. 

I added this call just to set the object_size in the request object to size given in Content-Length (which would be 0). We may need this for verifications or other decision making.